c_crypto.h

```diff
@@ -3,59 +3,44 @@
 
 #include <openssl/aes.h>
 #include <openssl/bio.h>
+#include <openssl/crypto.h>
 #include <openssl/err.h>
 #include <openssl/evp.h>
 #include <openssl/hmac.h>
+#include <openssl/md5.h>
 #include <openssl/pem.h>
 #include <openssl/rand.h>
 #include <openssl/rsa.h>
 #include <openssl/sha.h>
 #include <openssl/ssl.h>
 #include <openssl/x509.h>
 
-#include "c_common.h"
-
-#define AES_CBC_128_KEY_SIZE 128
-#define AES_CBC_128_IV_SIZE 16
-#define DIST_KEY_EXPIRATION_TIME_SIZE 6
-#define KEY_EXPIRATION_TIME_SIZE 6
-#define SESSION_KEY_ID_SIZE 8
+#define AES_128_KEY_SIZE_IN_BYTES 16
+#define AES_128_IV_SIZE 16
+#define AES_128_CBC_IV_SIZE 16
+#define AES_128_CTR_IV_SIZE 16
+#define AES_128_GCM_IV_SIZE 12
+#define AES_GCM_TAG_SIZE 12
 #define ABS_VALIDITY_SIZE 6
 #define REL_VALIDITY_SIZE 6
-#define MAC_KEY_SIZE 32  // FIXME: To be replaced by config.
 #define MAX_MAC_KEY_SIZE 32
+#define MAC_KEY_SHA256_SIZE 32
 #define CIPHER_KEY_SIZE 16  // FIXME: To be replaced by config.
-#define MAX_CIPHER_KEY_SIZE 32
 #define RSA_KEY_SIZE 256
 #define RSA_ENCRYPT_SIGN_SIZE RSA_KEY_SIZE * 2
 #define SHA256_DIGEST_LENGTH 32
 
+// Encryption Mode //
+#define AES_128_CBC 101
+#define AES_128_CTR 102
+#define AES_128_GCM 103
```

Comment:
Please change the encryption modes to enum.

Reply:
Fixed 27946ce62ef4801134ca26b71c1f488e3a2b0d76.